### ISSUE TITLE
Controller bombed with mutate/validate webhook events

### ISSUE THREAD

**Issue by Diontsoumas** [actions-runner-controller/actions-runner-controller#1643]
### Controller Version

0.25.1

### Helm Chart Version

0.20.1

### CertManager Version

1.8.1

### Deployment Method

Helm

### cert-manager installation

We had already installed cert-manager separately from ARC through its official Helm chart.

### Checks

- [X] This isn't a question or user support case (For Q&A and community support, go to [Discussions](https://github.com/actions-runner-controller/actions-runner-controller/discussions). It might also be a good idea to contract with any of contributors and maintainers if your business is so critical and therefore you need priority support
- [X] I've read [releasenotes](https://github.com/actions-runner-controller/actions-runner-controller/tree/master/docs/releasenotes) before submitting this issue and I'm sure it's not due to any recently-introduced backward-incompatible changes
- [X] My actions-runner-controller version (v0.x.y) does support the feature
- [X] I've already upgraded ARC (including the CRDs, see charts/actions-runner-controller/docs/UPGRADING.md for details) to the latest and it didn't fix the issue

### Resource Definitions

```yaml
apiVersion: actions.summerwind.dev/v1alpha1
kind: RunnerDeployment
metadata:
  name: ga-runner-deployment-quick
  namespace: system-actions-runner
spec:
  template:
    spec:
      labels:
        - quick
        - gwi-hosted
      organization: <hidden>
      # Run single container
      dockerdWithinRunnerContainer: true
      image: summerwind/actions-runner-dind:v2.294.0-ubuntu-20.04-e3deb0d
      env: []
      tolerations:
      - effect: NoSchedule
        key: dedicated
        operator: Equal
        value: github_actions
      - effect: NoSchedule
        key: type
        operator: Equal
        value: quick
      nodeSelector:
        role: github-actions-runners
        type: quick
      resources:
        limits:
          cpu: 2
          memory: 4Gi
        requests:
          cpu: 2
          memory: 4Gi

---

apiVersion: actions.summerwind.dev/v1alpha1
kind: RunnerDeployment
metadata:
  name: ga-runner-deployment-default
  namespace: system-actions-runner
spec:
  template:
    spec:
      labels:
        - gwi-hosted
      organization: <hidden>
      # Run single container
      dockerdWithinRunnerContainer: true
      image: summerwind/actions-runner-dind:v2.294.0-ubuntu-20.04-e3deb0d
      env: []
      tolerations:
      - effect: NoSchedule
        key: dedicated
        operator: Equal
        value: github_actions
      - effect: NoSchedule
        key: type
        operator: Equal
        value: default
      nodeSelector:
        role: github-actions-runners
        type: default
      resources:
        limits:
          cpu: 3
          memory: 6Gi
        requests:
          cpu: 3
          memory: 6Gi


---

apiVersion: actions.summerwind.dev/v1alpha1
kind: RunnerDeployment
metadata:
  name: ga-runner-deployment-performance
  namespace: system-actions-runner
spec:
  template:
    spec:
      labels:
        - performance
        - gwi-hosted
      organization: <hidden>
      # Run single container
      dockerdWithinRunnerContainer: true
      image: summerwind/actions-runner-dind:v2.294.0-ubuntu-20.04-e3deb0d
      env: []
      tolerations:
      - effect: NoSchedule
        key: dedicated
        operator: Equal
        value: github_actions
      - effect: NoSchedule
        key: type
        operator: Equal
        value: performance
      nodeSelector:
        role: github-actions-runners
        type: performance
      resources:
        limits:
          cpu: 4
          memory: 10Gi
        requests:
          cpu: 4
          memory: 10Gi

---

apiVersion: actions.summerwind.dev/v1alpha1
kind: HorizontalRunnerAutoscaler
metadata:
  namespace: system-actions-runner
  name: ga-runner-deploy-autoscaler-quick
spec:
  scaleTargetRef:
    name: ga-runner-deployment-quick
  scaleUpTriggers:
  - githubEvent:
      workflowJob: {}
    amount: 1
    duration: "5m"
  minReplicas: 2
  maxReplicas: 30

---

apiVersion: actions.summerwind.dev/v1alpha1
kind: HorizontalRunnerAutoscaler
metadata:
  namespace: system-actions-runner
  name: ga-runner-deploy-autoscaler-default
spec:
  scaleTargetRef:
    name: ga-runner-deployment-default
  scaleUpTriggers:
  - githubEvent:
      workflowJob: {}
    amount: 1
    duration: "5m"
  minReplicas: 2
  maxReplicas: 30

---

apiVersion: actions.summerwind.dev/v1alpha1
kind: HorizontalRunnerAutoscaler
metadata:
  namespace: system-actions-runner
  name: ga-runner-deploy-autoscaler-performance
spec:
  scaleTargetRef:
    name: ga-runner-deployment-performance
  scaleUpTriggers:
  - githubEvent:
      workflowJob: {}
    amount: 1
    duration: "5m"
  minReplicas: 2
  maxReplicas: 30
```

### To Reproduce

```markdown
1. Deploy the latest version through Helm chart.
2. Make sure that the RunnerDeployment and HorizontalRunnerAutoscaler resources have been created correctly.
3. Send some traffic to those pods
4. Check the controller logs `kubectl logs -f <pod-id> manager`
```

### Describe the bug

Controller gets bombed with info and debug statements, maybe a few thousands lines of evens per second. No errors with a quick glance (code is 200), but the same RunnerDeployment pods trigger the same requests again and again, a few hundred of times per second. This can't be right, and is a different behaviour from Chart v0.18 (controller v0.23.0) that we were previously trying.

If this is the correct behaviour with just a few pods and no traffic, how will the system behave when it scales with many pipelines?

### Describe the expected behavior

This is a fresh installation to our production systems. Previously, we were testing waters on a testing cluster using Chart v0.18 (controller v.0.23.0), which didn't have the same behaviour. Attached logs for both controller versions.

### Controller Logs

```shell
Controller logs for 0.25.1: (logs cover the first ~1.30' of the pod creation in order to make the gist smaller to read ): https://gist.github.com/Diontsoumas/ecca7ad85648c0bffe9c3980e166eb49#file-controller-logs-0-20

Controller logs for 0.23.0:
https://gist.github.com/Diontsoumas/ecca7ad85648c0bffe9c3980e166eb49#file-controller-logs-0-18
```

### Runner Pod Logs

```shell
N/A
```

### Additional Context

N/A

**Comment by mumoshu** [comment]
@Diontsoumas Hey! Thanks for reporting.
Unfortunately, I couldn't reproduce this.

Can you clarify a bit more on the `1. Deploy the latest version through Helm chart.` part?
What's the exact command you run and your values.yaml?
What's your K8s version?

Also, at glance, I see a lot of mutating and validating webhook responses in your logs. They do happen every time any K8s client or ARC updates any ARC-related custom resource(including Runner, RunnerSet, RunnerDeployment, and so on).
But the fact that I only see repeating mutating/validating webhook execution logs only for two RunnerReplicaSets, `ga-runner-deployment-quick-4vls7` and `ga-runner-deployment-performance-d5hd`, even though you have other RunnerDeployment named `ga-runner-deployment-default`, is redherring.

What do you see if you removed all the HRAs and RDs related to the `quick` and `performance` runners? Does the issue reproduce when you only have either `performance` or `quick` configurations, without the `default` configuration?

**Comment by Diontsoumas** [comment]
Thank you for the fast response @mumoshu! Kubernetes version is 1.21, on GKE. We use the Helm provider through Terraform:

``
resource "helm_release" "actions_runner_controller" {
  name             = "actions-runner-controller"
  repository       = "https://actions-runner-controller.github.io/actions-runner-controller"
  namespace        = system-actions-runner
  chart            = "actions-runner-controller"
  version          = "0.20.1"
  cleanup_on_fail  = true
  atomic           = true
  lint             = true
  timeout          = "300"
  create_namespace = false
  values = [
    "${file("controller-values.yaml")}",
  ]
}
``

The controller-values.yaml is the following:

``
githubWebhookServer:
  enabled: true
  service:
    type: NodePort
  ingress:
    enabled: true
    annotations:
      nginx.ingress.kubernetes.io/ssl-redirect: "true"
      cert-manager.io/cluster-issuer: letsencrypt-production
    ingressClassName: nginx-external
    hosts:
      - host: <hidden>
        paths:
          - path: "/"
            pathType: ImplementationSpecific
    tls:
    - secretName: ga-external-tls
      hosts:
        - <hidden>

metrics:
  serviceMonitor: true

labels:
  external-dns/create-records: '"yes"'
``

Both the terraform block and the values are trivial. I will run the tests that you asked and will report shortly :)

**Comment by mumoshu** [comment]
> Controller logs for 0.20

> Controller logs for 0.18

Are those really controller versions?

Both of those seems to conflict with what you've written in the form 🤔 

> Controller Version
> 0.25.1

**Comment by mumoshu** [comment]
@Diontsoumas Are there other custom K8s controllers and operators in your cluster? One of the possible causes of issues like this one is that some other controller is continuously mutating runnerreplicasets for whatever reason I don't know.

**Comment by mumoshu** [comment]
@Diontsoumas Hey! Two questions if you don't mind. I'm really confused how issues like this can happen.

> Previously, we were testing waters on a testing cluster using v0.18

Assuming it's the chart version, does the chart of ver 0.20.1 work on the testing cluster? What's the difference between the testing and the production clusters?

**Comment by Diontsoumas** [comment]
No you are right, the controller logs point to the Helm version (I will edit my post to update the info). So to clarify things:

- We were testing Chart 0.18 -> controller `v0.23.0`
- Production now runs Chart 0.20.1 -> controller `v0.25.1`

**Test 1**: Multiple RunnerDeployments `ga-runner-deployment-default`, `ga-runner-deployment-quick` **without** HRD
Result: Controller logs look normal. No much logs, since HRD is not in place.

```
2022-07-13T11:33:23Z	DEBUG	actions-runner-controller.runner	Runner appears to have been registered and running.	{"runner": "system-actions-runner/ga-runner-deployment-quick-hbbm6-sb9b4", "podCreationTimestamp": "2022-07-13 11:33:21 +0000 UTC"}
```

**Test 2**: Multiple RunnerDeployments `ga-runner-deployment-default`, `ga-runner-deployment-quick` **with** single HRD on `ga-runner-deployment-default`

Before triggering pipelines, everything looks normal:
```
2022-07-13T11:47:38Z	DEBUG	actions-runner-controller.runner	Runner appears to have been registered and running.	{"runner": "system-actions-runner/ga-runner-deployment-default-hzrmm-w777j", "podCreationTimestamp": "2022-07-13 11:47:37 +0000 UTC"}
2022-07-13T11:48:30Z	DEBUG	actions-runner-controller.horizontalrunnerautoscaler	Calculated desired replicas of 2	{"horizontalrunnerautoscaler": "system-actions-runner/ga-runner-deploy-autoscaler-default", "suggested": 2, "reserved": 0, "min": 2, "max": 30, "last_scale_up_time": "2022-07-13 11:47:36 +0000 UTC", "scale_down_delay_until": "2022-07-13T11:57:36Z"}
```

After triggering a single pipeline, which sites on the `quick` pod, everything goes crazy. Interestingly, requests display the `default` pods now (the ones that `HorizontalRunnerAutoscaler` scales)

```
2022-07-13T11:49:10Z	DEBUG	actions-runner-controller.horizontalrunnerautoscaler	Calculated desired replicas of 3	{"horizontalrunnerautoscaler": "system-actions-runner/ga-runner-deploy-autoscaler-default", "suggested": 2, "reserved": 1, "min": 2, "max": 30}
2022-07-13T11:49:10Z	DEBUG	controller-runtime.webhook.webhooks	received request{"webhook": "/mutate-actions-summerwind-dev-v1alpha1-runnerdeployment", "UID": "2bd08c9d-9554-4eff-8809-ec5e2fef7bea", "kind": "actions.summerwind.dev/v1alpha1, Kind=RunnerDeployment", "resource": {"group":"actions.summerwind.dev","version":"v1alpha1","resource":"runnerdeployments"}}
2022-07-13T11:49:10Z	DEBUG	controller-runtime.webhook.webhooks	wrote response	{"webhook": "/mutate-actions-summerwind-dev-v1alpha1-runnerdeployment", "code": 200, "reason": "", "UID": "2bd08c9d-9554-4eff-8809-ec5e2fef7bea", "allowed": true}
2022-07-13T11:49:10Z	DEBUG	controller-runtime.webhook.webhooks	received request{"webhook": "/validate-actions-summerwind-dev-v1alpha1-runnerdeployment", "UID": "5808286a-319e-4261-aa08-6c06047535e5", "kind": "actions.summerwind.dev/v1alpha1, Kind=RunnerDeployment", "resource": {"group":"actions.summerwind.dev","version":"v1alpha1","resource":"runnerdeployments"}}
2022-07-13T11:49:10Z	INFO	runnerdeployment-resource	validate resource to be updated	{"name": "ga-runner-deployment-default"}
2022-07-13T11:49:10Z	DEBUG	controller-runtime.webhook.webhooks	wrote response	{"webhook": "/validate-actions-summerwind-dev-v1alpha1-runnerdeployment", "code": 200, "reason": "", "UID": "5808286a-319e-4261-aa08-6c06047535e5", "allowed": true}
2022-07-13T11:49:10Z	DEBUG	actions-runner-controller.horizontalrunnerautoscaler	Calculated desired replicas of 3	{"horizontalrunnerautoscaler": "system-actions-runner/ga-runner-deploy-autoscaler-default", "suggested": 2, "reserved": 1, "min": 2, "max": 30, "last_scale_up_time": "2022-07-13 11:49:10 +0000 UTC", "scale_down_delay_until": "2022-07-13T11:59:10Z"}
```

So it's clear that the `HorizontalRunnerAutoscaler` is somehow involved.

**Test 3**: Single RunnerDeployment `ga-runner-deployment-default` **with** single HRD on `ga-runner-deployment-default`

Results are the same, logs are getting crazy.

### Next test is to install helm v0.28 (controller v0.23.0) to production cluster and rerun all the above.

To answer your questions, the two clusters shouldn't be much different. Both of them have things like the Prometheus operator or the nginx Ingress controller. Is there a way to check for older leftover controllers?

**Comment by Diontsoumas** [comment]
Things look much more stable with controller v0.23.0. All 3 `RunnerDeployments` with their own `HorizontalRunnerAutoscalers` on the production cluster look fine in logs. Every minute or so, I see:

```
2022-07-13T12:42:05Z	DEBUG	actions-runner-controller.horizontalrunnerautoscaler	Calculated desired replicas of 3	{"horizontalrunnerautoscaler": "system-actions-runner/ga-runner-deploy-autoscaler-performance", "suggested": 2, "reserved": 1, "min": 2, "max": 30, "last_scale_up_time": "2022-07-13 12:38:42 +0000 UTC", "scale_down_delay_until": "2022-07-13T12:48:42Z"}
2022-07-13T12:42:05Z	DEBUG	controller-runtime.webhook.webhooks	received request{"webhook": "/mutate-actions-summerwind-dev-v1alpha1-runnerdeployment", "UID": "481bfc99-cb58-4f8f-9480-e33535376001", "kind": "actions.summerwind.dev/v1alpha1, Kind=RunnerDeployment", "resource": {"group":"actions.summerwind.dev","version":"v1alpha1","resource":"runnerdeployments"}}
2022-07-13T12:42:05Z	DEBUG	controller-runtime.webhook.webhooks	wrote response	{"webhook": "/mutate-actions-summerwind-dev-v1alpha1-runnerdeployment", "code": 200, "reason": "", "UID": "481bfc99-cb58-4f8f-9480-e33535376001", "allowed": true}
2022-07-13T12:42:05Z	DEBUG	controller-runtime.webhook.webhooks	received request{"webhook": "/validate-actions-summerwind-dev-v1alpha1-runnerdeployment", "UID": "c7d4fc9d-2360-4ad9-b0d7-269113575f8a", "kind": "actions.summerwind.dev/v1alpha1, Kind=RunnerDeployment", "resource": {"group":"actions.summerwind.dev","version":"v1alpha1","resource":"runnerdeployments"}}
2022-07-13T12:42:05Z	INFO	runnerdeployment-resource	validate resource to be updated	{"name": "ga-runner-deployment-performance"}
2022-07-13T12:42:05Z	DEBUG	controller-runtime.webhook.webhooks	wrote response	{"webhook": "/validate-actions-summerwind-dev-v1alpha1-runnerdeployment", "code": 200, "reason": "", "UID": "c7d4fc9d-2360-4ad9-b0d7-269113575f8a", "allowed": true}
2022-07-13T12:42:05Z	DEBUG	actions-runner-controller.horizontalrunnerautoscaler	Calculated desired replicas of 2	{"horizontalrunnerautoscaler": "system-actions-runner/ga-runner-deploy-autoscaler-default", "suggested": 2, "reserved": 0, "min": 2, "max": 30, "last_scale_up_time": "2022-07-13 12:38:14 +0000 UTC", "scale_down_delay_until": "2022-07-13T12:48:14Z"}
2022-07-13T12:42:05Z	DEBUG	actions-runner-controller.horizontalrunnerautoscaler	Calculated desired replicas of 2	{"horizontalrunnerautoscaler": "system-actions-runner/ga-runner-deploy-autoscaler-quick", "suggested": 2, "reserved": 0, "min": 2, "max": 30, "last_scale_up_time": "2022-07-13 12:38:14 +0000 UTC", "scale_down_delay_until": "2022-07-13T12:48:14Z"}
```

I run 2-3 pipelines to confirm that things are stable. Can this somehow be a side effect of [Webhook-based scaling is even more reliable](https://github.com/actions-runner-controller/actions-runner-controller/blob/master/docs/releasenotes/0.25.md#fix--webhook-based-scaling-is-even-more-reliable) fix in v0.25.1?

**Comment by zotrix** [comment]
Same issue

```
{"level":"info","ts":"2022-07-14T09:20:16Z","logger":"runnerreplicaset-resource","msg":"validate resource to be updated","name":"resource-n8xvz"}
{"level":"info","ts":"2022-07-14T09:20:16Z","logger":"runnerreplicaset-resource","msg":"validate resource to be updated","name":"resource-tools-phlrd"}
{"level":"info","ts":"2022-07-14T09:20:16Z","logger":"runnerreplicaset-resource","msg":"validate resource to be updated","name":"resource-large-4bfvk"}
{"level":"info","ts":"2022-07-14T09:20:16Z","logger":"runnerreplicaset-resource","msg":"validate resource to be updated","name":"resource-n8xvz"}
{"level":"info","ts":"2022-07-14T09:20:16Z","logger":"runnerreplicaset-resource","msg":"validate resource to be updated","name":"resource-tools-phlrd"}
{"level":"info","ts":"2022-07-14T09:20:16Z","logger":"runnerreplicaset-resource","msg":"validate resource to be updated","name":"resource-large-4bfvk"}
{"level":"info","ts":"2022-07-14T09:20:16Z","logger":"runnerreplicaset-resource","msg":"validate resource to be updated","name":"resource-n8xvz"}
{"level":"info","ts":"2022-07-14T09:20:16Z","logger":"runnerreplicaset-resource","msg":"validate resource to be updated","name":"resource-tools-phlrd"}
{"level":"info","ts":"2022-07-14T09:20:17Z","logger":"runnerreplicaset-resource","msg":"validate resource to be updated","name":"resource-large-4bfvk"}
{"level":"info","ts":"2022-07-14T09:20:17Z","logger":"runnerreplicaset-resource","msg":"validate resource to be updated","name":"resource-n8xvz"}
{"level":"info","ts":"2022-07-14T09:20:17Z","logger":"runnerreplicaset-resource","msg":"validate resource to be updated","name":"resource-tools-phlrd"}
{"level":"info","ts":"2022-07-14T09:20:17Z","logger":"runnerreplicaset-resource","msg":"validate resource to be updated","name":"resource-large-4bfvk"}
{"level":"info","ts":"2022-07-14T09:20:17Z","logger":"runnerreplicaset-resource","msg":"validate resource to be updated","name":"resource-n8xvz"}
{"level":"info","ts":"2022-07-14T09:20:17Z","logger":"runnerreplicaset-resource","msg":"validate resource to be updated","name":"resource-tools-phlrd"}
{"level":"info","ts":"2022-07-14T09:20:17Z","logger":"runnerreplicaset-resource","msg":"validate resource to be updated","name":"resource-large-4bfvk"}
{"level":"info","ts":"2022-07-14T09:20:17Z","logger":"runnerreplicaset-resource","msg":"validate resource to be updated","name":"resource-n8xvz"}
{"level":"info","ts":"2022-07-14T09:20:17Z","logger":"runnerreplicaset-resource","msg":"validate resource to be updated","name":"resource-tools-phlrd"}
{"level":"info","ts":"2022-07-14T09:20:17Z","logger":"runnerreplicaset-resource","msg":"validate resource to be updated","name":"resource-large-4bfvk"}
{"level":"info","ts":"2022-07-14T09:20:17Z","logger":"runnerreplicaset-resource","msg":"validate resource to be updated","name":"resource-n8xvz"}
{"level":"info","ts":"2022-07-14T09:20:17Z","logger":"runnerreplicaset-resource","msg":"validate resource to be updated","name":"resource-tools-phlrd"}
{"level":"info","ts":"2022-07-14T09:20:17Z","logger":"runnerreplicaset-resource","msg":"validate resource to be updated","name":"resource-large-4bfvk"}
{"level":"info","ts":"2022-07-14T09:20:17Z","logger":"runnerreplicaset-resource","msg":"validate resource to be updated","name":"resource-n8xvz"}
{"level":"info","ts":"2022-07-14T09:20:17Z","logger":"runnerreplicaset-resource","msg":"validate resource to be updated","name":"resource-tools-phlrd"}
{"level":"info","ts":"2022-07-14T09:20:17Z","logger":"runnerreplicaset-resource","msg":"validate resource to be updated","name":"resource-large-4bfvk"}
{"level":"info","ts":"2022-07-14T09:20:17Z","logger":"runnerreplicaset-resource","msg":"validate resource to be updated","name":"resource-n8xvz"}
{"level":"info","ts":"2022-07-14T09:20:17Z","logger":"runnerreplicaset-resource","msg":"validate resource to be updated","name":"resource-tools-phlrd"}
{"level":"info","ts":"2022-07-14T09:20:17Z","logger":"runnerreplicaset-resource","msg":"validate resource to be updated","name":"resource-large-4bfvk"}
{"level":"info","ts":"2022-07-14T09:20:17Z","logger":"runnerreplicaset-resource","msg":"validate resource to be updated","name":"resource-n8xvz"}
{"level":"info","ts":"2022-07-14T09:20:18Z","logger":"runnerreplicaset-resource","msg":"validate resource to be updated","name":"resource-tools-phlrd"}
{"level":"info","ts":"2022-07-14T09:20:18Z","logger":"runnerreplicaset-resource","msg":"validate resource to be updated","name":"resource-large-4bfvk"}
{"level":"info","ts":"2022-07-14T09:20:18Z","logger":"runnerreplicaset-resource","msg":"validate resource to be updated","name":"resource-n8xvz"}
{"level":"info","ts":"2022-07-14T09:20:18Z","logger":"runnerreplicaset-resource","msg":"validate resource to be updated","name":"resource-tools-phlrd"}
```

In GKE with version: 1.22.8-gke.202
arc installed via helm from 
```
    app.kubernetes.io/version: 0.25.1
    helm.sh/chart: actions-runner-controller-0.20.1
```

**Comment by zotrix** [comment]
@mumoshu reproduced with update 
```
-github_runner_controller_version: 0.18.0
+github_runner_controller_version: 0.20.1
``

before enable github webhook logs looks normal, but after enabling I see this in logs

```
{"level":"debug","ts":"2022-07-14T09:57:20Z","logger":"controller-runtime.webhook.webhooks","msg":"received request","webhook":"/mutate-actions-summerwind-dev-v1alpha1-runnerreplicaset","UID":"36dfb3c5-491c-477d-ad5e-0a35d4f543f4","kind":"actions.summerwind.dev/v1alpha1, Kind=RunnerReplicaSet","resource":{"group":"actions.summerwind.dev","version":"v1alpha1","resource":"runnerreplicasets"}}
{"level":"debug","ts":"2022-07-14T09:57:20Z","logger":"controller-runtime.webhook.webhooks","msg":"wrote response","webhook":"/mutate-actions-summerwind-dev-v1alpha1-runnerreplicaset","code":200,"reason":"","UID":"36dfb3c5-491c-477d-ad5e-0a35d4f543f4","allowed":true}
{"level":"debug","ts":"2022-07-14T09:57:20Z","logger":"controller-runtime.webhook.webhooks","msg":"received request","webhook":"/validate-actions-summerwind-dev-v1alpha1-runnerreplicaset","UID":"06dc355e-ae6a-49ad-8618-71e1adf0cc22","kind":"actions.summerwind.dev/v1alpha1, Kind=RunnerReplicaSet","resource":{"group":"actions.summerwind.dev","version":"v1alpha1","resource":"runnerreplicasets"}}
{"level":"info","ts":"2022-07-14T09:57:20Z","logger":"runnerreplicaset-resource","msg":"validate resource to be updated","name":"arc-job-dev-gwrkq"}
{"level":"debug","ts":"2022-07-14T09:57:20Z","logger":"controller-runtime.webhook.webhooks","msg":"wrote response","webhook":"/validate-actions-summerwind-dev-v1alpha1-runnerreplicaset","code":200,"reason":"","UID":"06dc355e-ae6a-49ad-8618-71e1adf0cc22","allowed":true}
{"level":"debug","ts":"2022-07-14T09:57:20Z","logger":"controller-runtime.webhook.webhooks","msg":"received request","webhook":"/mutate-actions-summerwind-dev-v1alpha1-runnerreplicaset","UID":"4167028a-930a-4e75-809e-ebde279742fb","kind":"actions.summerwind.dev/v1alpha1, Kind=RunnerReplicaSet","resource":{"group":"actions.summerwind.dev","version":"v1alpha1","resource":"runnerreplicasets"}}
{"level":"debug","ts":"2022-07-14T09:57:20Z","logger":"controller-runtime.webhook.webhooks","msg":"wrote response","webhook":"/mutate-actions-summerwind-dev-v1alpha1-runnerreplicaset","code":200,"reason":"","UID":"4167028a-930a-4e75-809e-ebde279742fb","allowed":true}
{"level":"debug","ts":"2022-07-14T09:57:20Z","logger":"controller-runtime.webhook.webhooks","msg":"received request","webhook":"/validate-actions-summerwind-dev-v1alpha1-runnerreplicaset","UID":"d3315491-5e67-43ca-9154-8ce3718d3bf6","kind":"actions.summerwind.dev/v1alpha1, Kind=RunnerReplicaSet","resource":{"group":"actions.summerwind.dev","version":"v1alpha1","resource":"runnerreplicasets"}}
{"level":"info","ts":"2022-07-14T09:57:20Z","logger":"runnerreplicaset-resource","msg":"validate resource to be updated","name":"arc-job-dev-gwrkq"}
{"level":"debug","ts":"2022-07-14T09:57:20Z","logger":"controller-runtime.webhook.webhooks","msg":"wrote response","webhook":"/validate-actions-summerwind-dev-v1alpha1-runnerreplicaset","code":200,"reason":"","UID":"d3315491-5e67-43ca-9154-8ce3718d3bf6","allowed":true}
{"level":"debug","ts":"2022-07-14T09:57:20Z","logger":"controller-runtime.webhook.webhooks","msg":"received request","webhook":"/mutate-actions-summerwind-dev-v1alpha1-runnerreplicaset","UID":"44a545bf-8101-4aee-82d1-7e2f758b8cb3","kind":"actions.summerwind.dev/v1alpha1, Kind=RunnerReplicaSet","resource":{"group":"actions.summerwind.dev","version":"v1alpha1","resource":"runnerreplicasets"}}
{"level":"debug","ts":"2022-07-14T09:57:20Z","logger":"controller-runtime.webhook.webhooks","msg":"wrote response","webhook":"/mutate-actions-summerwind-dev-v1alpha1-runnerreplicaset","code":200,"reason":"","UID":"44a545bf-8101-4aee-82d1-7e2f758b8cb3","allowed":true}
{"level":"debug","ts":"2022-07-14T09:57:20Z","logger":"controller-runtime.webhook.webhooks","msg":"received request","webhook":"/validate-actions-summerwind-dev-v1alpha1-runnerreplicaset","UID":"c61e2956-6e98-4086-a626-8a01b3198a7d","kind":"actions.summerwind.dev/v1alpha1, Kind=RunnerReplicaSet","resource":{"group":"actions.summerwind.dev","version":"v1alpha1","resource":"runnerreplicasets"}}
{"level":"info","ts":"2022-07-14T09:57:20Z","logger":"runnerreplicaset-resource","msg":"validate resource to be updated","name":"arc-job-dev-gwrkq"}
{"level":"debug","ts":"2022-07-14T09:57:20Z","logger":"controller-runtime.webhook.webhooks","msg":"wrote response","webhook":"/validate-actions-summerwind-dev-v1alpha1-runnerreplicaset","code":200,"reason":"","UID":"c61e2956-6e98-4086-a626-8a01b3198a7d","allowed":true}```

**Comment by mumoshu** [comment]
@Diontsoumas Thanks a lot for the update! I was able to reproduce your issue, and #1650 is the fix that worked for me.

**Comment by Diontsoumas** [comment]
@mumoshu we did some more digging around and we think it's the manager that causes this loophole.

According to the logs, it tries constantly to mutate and validate the `runnerreplicaset` crd. We can verify this if we describe the resource in question. See the value of the generation metadata below.

```
apiVersion: actions.summerwind.dev/v1alpha1
kind: RunnerReplicaSet
metadata:
  creationTimestamp: "2022-07-14T10:44:23Z"
  generateName: ga-runner-deployment-default-test-
  generation: 50597
```

We think it's the manager that causes the loop, because if we update the `actions-runner-controller-mutating-webhook-configuration` (the mutation webhook), and just remove the `UPDATE` operation on the `runnerreplicasets` resource, then the log flow stops.

We believe that if it was another controller causing the issue, at least the `validate` webhook or `CREATE` on the `mutate` would be triggered.

By the way, we also tested controller 0.25.1 on our testing cluster and can confirm the same behaviour.

**Comment by mumoshu** [comment]
@Diontsoumas Your finding definitely looks like another data point that validates the issue and my fix 😄 Thanks for double-checking!

**Comment by zotrix** [comment]
@mumoshu  I try to reproduce in development env

> @Diontsoumas Thanks a lot for the update! I was able to reproduce your issue, and #1650 is the fix that worked for me.

thanks, this fix logging for me also

**Comment by KENNYSOFT** [comment]
Thanks for the fix! This was why I reverted to chart 0.19.1 (app 0.24.1) but I've missed reporting.
